### PR TITLE
Simplify the Envoy integration by using the new rules_fuzzing primitives.

### PR DESCRIPTION
**Pull request by stefanbucur**
We should wait for envoyproxy/envoy#14834 to be submitted first. CC @asraa @htuch

### CODE REVIEW

**Review by htuch — COMMENTED** [review]
LGTM (but wait until dependent PR is merged). This is super awesome, so much simpler!

File: projects/envoy/Dockerfile
Comment (inferno-chromium) [inline projects/envoy/Dockerfile:32]: Will six be needed just for envoy or all projects ?

**Review by inferno-chromium — COMMENTED** [review]
Can you please check build failures

```
+ FUZZING_CXXFLAGS='-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++'
+ bazel build --verbose_failures --dynamic_mode=off --spawn_strategy=standalone --genrule_strategy=standalone '--local_cpu_resources=HOST_CPUS*0.32' --build_tag_filters=-no_asan --config=oss-fuzz --copt -D__SANITIZE_ADDRESS__ --copt -DADDRESS_SANITIZER=1 //test/server/config_validation:xds_fuzz_test_oss_fuzz //test/server/config_validation:config_fuzz_test_oss_fuzz //test/server:server_fuzz_test_oss_fuzz //test/integration:h2_capture_persistent_fuzz_test_oss_fuzz //test/integration:h2_capture_fuzz_test_oss_fuzz //test/integration:h2_capture_direct_response_persistent_fuzz_test_oss_fuzz //test/integration:h2_capture_direct_response_fuzz_test_oss_fuzz //test/integration:h1_capture_persistent_fuzz_test_oss_fuzz //test/integration:h1_capture_fuzz_test_oss_fuzz //test/integration:h1_capture_direct_response_persistent_fuzz_test_oss_fuzz //test/integration:h1_capture_direct_response_fuzz_test_oss_fuzz //test/extensions/tracers/xray:xray_fuzz_test_oss_fuzz //test/extensions/filters/udp/dns_filter:dns_filter_fuzz_test_oss_fuzz //test/extensions/filters/network/local_ratelimit:local_ratelimit_fuzz_test_oss_fuzz //test/extensions/filters/network/ext_authz:ext_authz_fuzz_test_oss_fuzz //test/extensions/filters/network/common/fuzz:network_writefilter_fuzz_test_oss_fuzz //test/extensions/filters/network/common/fuzz:network_readfilter_fuzz_test_oss_fuzz //test/extensions/filters/listener/tls_inspector:tls_inspector_fuzz_test_oss_fuzz //test/extensions/filters/listener/proxy_protocol:proxy_protocol_fuzz_test_oss_fuzz //test/extensions/filters/listener/original_src:original_src_fuzz_test_oss_fuzz //test/extensions/filters/listener/original_dst:original_dst_fuzz_test_oss_fuzz //test/extensions/filters/listener/http_inspector:http_inspector_fuzz_test_oss_fuzz //test/extensions/filters/http/common/fuzz:filter_fuzz_test_oss_fuzz //test/extensions/filters/http/cdn_loop:parser_fuzz_test_oss_fuzz //test/extensions/filters/common/expr:evaluator_fuzz_test_oss_fuzz //test/extensions/compression/gzip:compressor_fuzz_test_oss_fuzz //test/common/upstream:round_robin_load_balancer_fuzz_test_oss_fuzz //test/common/upstream:random_load_balancer_fuzz_test_oss_fuzz //test/common/upstream:least_request_load_balancer_fuzz_test_oss_fuzz //test/common/upstream:health_check_fuzz_test_oss_fuzz //test/common/stats:utility_fuzz_test_oss_fuzz //test/common/stats:symbol_table_fuzz_test_oss_fuzz //test/common/stats:stat_merger_fuzz_test_oss_fuzz //test/common/router:route_fuzz_test_oss_fuzz //test/common/router:header_parser_fuzz_test_oss_fuzz //test/common/network:utility_fuzz_test_oss_fuzz //test/common/json:json_fuzz_test_oss_fuzz //test/common/http/http2:response_header_fuzz_test_oss_fuzz //test/common/http/http2:request_header_fuzz_test_oss_fuzz //test/common/http/http2:hpack_fuzz_test_oss_fuzz //test/common/http:utility_fuzz_test_oss_fuzz //test/common/http:path_utility_fuzz_test_oss_fuzz //test/common/http:header_map_impl_fuzz_test_oss_fuzz //test/common/http:conn_manager_impl_fuzz_test_oss_fuzz //test/common/http:codec_impl_fuzz_test_oss_fuzz //test/common/grpc:codec_fuzz_test_oss_fuzz //test/common/formatter:substitution_formatter_fuzz_test_oss_fuzz //test/common/crypto:verify_signature_fuzz_test_oss_fuzz //test/common/crypto:get_sha_256_digest_fuzz_test_oss_fuzz //test/common/common:hash_fuzz_test_oss_fuzz //test/common/buffer:buffer_fuzz_test_oss_fuzz
ERROR: Config value 'oss-fuzz' is not defined in any .rc file
Building fuzzers failed.```

**Review by inferno-chromium — CHANGES_REQUESTED** [review]
See comment above

File: projects/envoy/Dockerfile
Comment (asraa) [inline projects/envoy/Dockerfile:35]: thanks for this! just realized bazel is already installed through bazelisk.

Comment (stefanbucur) [reply projects/envoy/Dockerfile:35]: Yeah, it was actually added pretty recently (#4976), in preparation for the Bazel support :)

File: projects/envoy/Dockerfile
Comment (stefanbucur) [reply projects/envoy/Dockerfile]: Good point - it is actually required by the Bazel rules for fuzzing. I've added it instead to the base image Dockerfile, PTAL.

**Comment by stefanbucur** [comment]
> Can you please check build failures
> 
> ```
> + FUZZING_CXXFLAGS='-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++'
> + bazel build --verbose_failures --dynamic_mode=off --spawn_strategy=standalone --genrule_strategy=standalone '--local_cpu_resources=HOST_CPUS*0.32' --build_tag_filters=-no_asan --config=oss-fuzz --copt -D__SANITIZE_ADDRESS__ --copt -DADDRESS_SANITIZER=1 //test/server/config_validation:xds_fuzz_test_oss_fuzz //test/server/config_validation:config_fuzz_test_oss_fuzz //test/server:server_fuzz_test_oss_fuzz //test/integration:h2_capture_persistent_fuzz_test_oss_fuzz //test/integration:h2_capture_fuzz_test_oss_fuzz //test/integration:h2_capture_direct_response_persistent_fuzz_test_oss_fuzz //test/integration:h2_capture_direct_response_fuzz_test_oss_fuzz //test/integration:h1_capture_persistent_fuzz_test_oss_fuzz //test/integration:h1_capture_fuzz_test_oss_fuzz //test/integration:h1_capture_direct_response_persistent_fuzz_test_oss_fuzz //test/integration:h1_capture_direct_response_fuzz_test_oss_fuzz //test/extensions/tracers/xray:xray_fuzz_test_oss_fuzz //test/extensions/filters/udp/dns_filter:dns_filter_fuzz_test_oss_fuzz //test/extensions/filters/network/local_ratelimit:local_ratelimit_fuzz_test_oss_fuzz //test/extensions/filters/network/ext_authz:ext_authz_fuzz_test_oss_fuzz //test/extensions/filters/network/common/fuzz:network_writefilter_fuzz_test_oss_fuzz //test/extensions/filters/network/common/fuzz:network_readfilter_fuzz_test_oss_fuzz //test/extensions/filters/listener/tls_inspector:tls_inspector_fuzz_test_oss_fuzz //test/extensions/filters/listener/proxy_protocol:proxy_protocol_fuzz_test_oss_fuzz //test/extensions/filters/listener/original_src:original_src_fuzz_test_oss_fuzz //test/extensions/filters/listener/original_dst:original_dst_fuzz_test_oss_fuzz //test/extensions/filters/listener/http_inspector:http_inspector_fuzz_test_oss_fuzz //test/extensions/filters/http/common/fuzz:filter_fuzz_test_oss_fuzz //test/extensions/filters/http/cdn_loop:parser_fuzz_test_oss_fuzz //test/extensions/filters/common/expr:evaluator_fuzz_test_oss_fuzz //test/extensions/compression/gzip:compressor_fuzz_test_oss_fuzz //test/common/upstream:round_robin_load_balancer_fuzz_test_oss_fuzz //test/common/upstream:random_load_balancer_fuzz_test_oss_fuzz //test/common/upstream:least_request_load_balancer_fuzz_test_oss_fuzz //test/common/upstream:health_check_fuzz_test_oss_fuzz //test/common/stats:utility_fuzz_test_oss_fuzz //test/common/stats:symbol_table_fuzz_test_oss_fuzz //test/common/stats:stat_merger_fuzz_test_oss_fuzz //test/common/router:route_fuzz_test_oss_fuzz //test/common/router:header_parser_fuzz_test_oss_fuzz //test/common/network:utility_fuzz_test_oss_fuzz //test/common/json:json_fuzz_test_oss_fuzz //test/common/http/http2:response_header_fuzz_test_oss_fuzz //test/common/http/http2:request_header_fuzz_test_oss_fuzz //test/common/http/http2:hpack_fuzz_test_oss_fuzz //test/common/http:utility_fuzz_test_oss_fuzz //test/common/http:path_utility_fuzz_test_oss_fuzz //test/common/http:header_map_impl_fuzz_test_oss_fuzz //test/common/http:conn_manager_impl_fuzz_test_oss_fuzz //test/common/http:codec_impl_fuzz_test_oss_fuzz //test/common/grpc:codec_fuzz_test_oss_fuzz //test/common/formatter:substitution_formatter_fuzz_test_oss_fuzz //test/common/crypto:verify_signature_fuzz_test_oss_fuzz //test/common/crypto:get_sha_256_digest_fuzz_test_oss_fuzz //test/common/common:hash_fuzz_test_oss_fuzz //test/common/buffer:buffer_fuzz_test_oss_fuzz
> ERROR: Config value 'oss-fuzz' is not defined in any .rc file
> Building fuzzers failed.```
> ```

Yeah, these were happening on an Envoy version prior to envoyproxy/envoy#14834. This is now submitted and I've also rebased my PR, hopefully things will build properly now.

**Review by inferno-chromium — APPROVED** [review]

**Review by asraa — APPROVED** [review]

**Comment by stefanbucur** [comment]
This one seems to be running out of disk space. According to Github, their action runners provide only [14 GiB of disk space](https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources), which feel insufficient for building all Envoy fuzz tests today.

How should we proceed here? I can do my own manual builds of everything locally if that works.

**Comment by inferno-chromium** [comment]
> This one seems to be running out of disk space. According to Github, their action runners provide only [14 GiB of disk space](https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources), which feel insufficient for building all Envoy fuzz tests today.
> 
> How should we proceed here? I can do my own manual builds of everything locally if that works.

yes just locally verify one libfuzzer-address version and then ping on chat to merge.

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**
